eng/packages/http-client-csharp/generator/Azure.Generator/src/Providers/ResourceDataSerializationProvider.cs

```diff
@@ -0,0 +1,21 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using Microsoft.Generator.CSharp.ClientModel.Providers;
+using Microsoft.Generator.CSharp.Input;
+using Microsoft.Generator.CSharp.Providers;
+using System.IO;
+
+namespace Azure.Generator.Providers
+{
+    internal class ResourceDataSerializationProvider : MrwSerializationTypeDefinition
+    {
+        public ResourceDataSerializationProvider(InputModelType inputModel, ModelProvider modelProvider) : base(inputModel, modelProvider)
+        {
+        }
+
+        protected override string BuildName() => $"{base.BuildName()}Data";
+
+        protected override string BuildRelativeFilePath() => Path.Combine("src", "Generated", $"{Name}.Serialization.cs");
```

Comment:
Does this need to be overriden?

Reply:
do you suggest using visitor to change the file location?

Reply:
I think we will probably need more logic in this class, so it makes more sense to leave it as you have it now.